lib/THC/THCTensor.c

```diff
@@ -599,7 +599,7 @@ void THCudaTensor_freeCopyTo(THCState *state, THCudaTensor *self, THCudaTensor *
 static void THCudaTensor_rawInit(THCState *state, THCudaTensor *self)
 {
   self->refcount = 1;
-  self->storage = NULL;
+  self->storage = THCudaStorage_new(state);
```

Comment:
I'm a bit worried about changing the behaviour here. THCudaTensor_data, for example, behaves slightly different now for empty tensors. Previously it would return `nil`, now it returns a pointer value `NULL` (causing different behaviour in conditionals, for example). I haven't seen any other cases of changing behaviour, but that doesn't mean there aren't any.
Could a tensor also store a device ID? Then you wouldn't have to create an empty storage, right?